### PR TITLE
[risk=low][RW-9264]Added logic to check to see if billing information has been loaded.

### PR DESCRIPTION
**Pull request by evrii**
Description:

Added error handling for when the billing information can not be fetched.

I could not find a great way to automate testing this. If you modify getBillingAccountInfo in ui/src/app/utils/workbench-gapi-client.ts to always throw an error, you can observe the UI error handling.

```
  return new Promise<GoogleBillingAccountInfo>(() => {
    throw new Error('Ouch');
  });

```

https://user-images.githubusercontent.com/24514630/206731364-2675da56-a0a9-47ae-9eda-a4ae5857f8a4.mov

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 1]
```diff
@@ -482,65 +485,83 @@ export const WorkspaceEdit = fp.flow(
...
-          // the source of truth, Google
           this.setState((prevState) =>
             fp.set(
               ['workspace', 'billingAccountName'],
-              fetchedBillingInfo.billingAccountName,
+              this.props.workspace.billingAccountName,
               prevState
             )
           );
+          this.setState({ fetchBillingAccountError: true });
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:557]: I don't know why these setStates were run separately before, but it's good practice to combine these as much as possible, like
```
this.setState({ fetchBillingAccountError: true, billingAccountFetched: false });
```

File: ui/src/app/pages/workspace/workspace-edit-text.tsx [hunk 2]
```diff
@@ -17,6 +17,11 @@ export const toolTipTextDemographic =
   'sex, gender identity, sexual orientation, geography, disability status, access to care, ' +
   'education level, or income\n';
 
+export const tooltipTextBillingWarning =
+  'Unfortunately it is not possible to change the billing account for this workspace at this time.  ' +
+  'Please try again later.  If you continue to see this message after 5 minutes after creation, ' +
+  'please contact support.';
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit-text.tsx:23]: Could use `<SupportMailto/>` here?

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 3]
```diff
@@ -482,65 +485,83 @@ export const WorkspaceEdit = fp.flow(
...
+            this.setState((prevState) =>
+              fp.set(
+                ['workspace', 'billingAccountName'],
+                fetchedBillingInfo.billingAccountName,
+                prevState
+              )
+            );
+          }
+          this.setState({ billingAccountFetched: true });
+        } catch (e) {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:543]: it can be hard to follow when a try-block is so long.  Would it make sense to extract it to a new function?

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 4]
```diff
@@ -482,65 +485,83 @@ export const WorkspaceEdit = fp.flow(
...
           displayBillingAccounts.push({
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:544]: I don't think this is correct.  Why are we assuming that the workspace has a user-provided billing account?

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 5]
```diff
@@ -482,65 +485,83 @@ export const WorkspaceEdit = fp.flow(
...
-          // If the user has owner access on the workspace but does not have access to the billing account
-          // that it is attached to, keep the server's current value for billingAccountName and add a shim
-          // entry into billingAccounts so the dropdown entry is not empty.
-          //
-          // The server will not perform an updateBillingInfo call if the received billingAccountName
-          // is the same as what is currently stored.
-          //
-          // This can happen if a workspace is shared to another researcher as an owner.
+        try {
+          const fetchedBillingInfo = await getBillingAccountInfo(
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:489]: I'd prefer to localize the try/catch as much as possible.  This is the line causing the error, so can we scope it down to here only?

also, we may be able to use `Promise.catch()` here instead

File: ui/src/app/pages/workspace/workspace-edit-text.tsx (hunk 2 above)
Comment (evrii) [reply ui/src/app/pages/workspace/workspace-edit-text.tsx]: Good call. I added it to the message.

File: ui/src/app/pages/workspace/workspace-edit.tsx (hunk 3 above)
Comment (evrii) [reply ui/src/app/pages/workspace/workspace-edit.tsx]: I gave it a shot. The logic in this try block was updating an array and  updating state. It seemed like more than I wanted to have in a single function. Please let me know if you think this can be improved.

File: ui/src/app/pages/workspace/workspace-edit.tsx (hunk 5 above)
Comment (evrii) [reply ui/src/app/pages/workspace/workspace-edit.tsx]: I switched this to Promise.catch instead of try/catch. Does the current approach seem better?

**Review by jmthibault79 — APPROVED** [review]
Tested locally by editing and saving workspaces successfully, with and without the promise-rejection hack.

**Comment by jmthibault79** [comment]
wait... the e2e failure seems to indicate the problem has not actually been resolved.  Please look into this before merging @evrii

**Review by jmthibault79 — CHANGES_REQUESTED** [review]
While the change looks reasonable, it does not resolve the issue

File: ui/src/app/utils/workbench-gapi-client.ts [hunk 6]
```diff
@@ -23,15 +23,17 @@ export async function getBillingAccountInfo(googleProject: string) {
...
         });
       } else {
         gapi.client.load('cloudbilling', 'v1', () => {
           gapi.client.cloudbilling.projects
             .getBillingInfo({
               name: 'projects/' + googleProject,
             })
-            .then((response) => resolve(JSON.parse(response.body)));
+            .then((response) => resolve(JSON.parse(response.body)))
+            .catch((error) => reject(error));
```
Comment (jmthibault79) [inline ui/src/app/utils/workbench-gapi-client.ts:36]: this is the line that fixed it for me.  The exception was not able to be handled because this was skipped

**Review by jmthibault79 — APPROVED** [review]
